pkg/neorpc/errors.go

```diff
@@ -30,33 +30,95 @@ const (
+	ErrUnableToVerify = NewSubmitError(-503, "Transaction already exists in the memory pool")
 	// ErrValidationFailed represents SubmitError with code -504.
-	ErrValidationFailed = NewSubmitError(-504, "Block or transaction validation failed.")
+	ErrValidationFailed = NewSubmitError(-504, "Validation failed")
 	// ErrPolicyFail represents SubmitError with code -505.
-	ErrPolicyFail = NewSubmitError(-505, "One of the Policy filters failed.")
-	// ErrUnknown represents SubmitError with code -500.
-	ErrUnknown = NewSubmitError(-500, "Unknown error.")
+	ErrPolicyFail = NewSubmitError(-505, "One of the Policy filters failed")
+	// ErrInvalidScript represents SubmitError with code -506.
+	ErrInvalidScript = NewSubmitError(-506, "Invalid script")
+	// ErrTransactionAttr represents SubmitError with code -507.
+	ErrTransactionAttr = NewSubmitError(-507, "Invalid transaction attribute")
+	// ErrInvalidSignature represents SubmitError with code -508.
+	ErrInvalidSignature = NewSubmitError(-508, "Invalid signature")
+	// ErrInvalidSize represents SubmitError with code -509.
+	ErrInvalidSize = NewSubmitError(-509, "Invalid inventory size")
+	// ErrExpiredTransaction represents SubmitError with code -510.
+	ErrExpiredTransaction = NewSubmitError(-510, "Expired transaction")
+	// ErrInsufficientFunds represents SubmitError with code -511.
+	ErrInsufficientFunds = NewSubmitError(-511, "Insufficient funds")
+
+	// ErrOracleNotRun represents SubmitError with code -601.
+	ErrOracleNotRun = NewSubmitError(-601, "Oracle service is not running")
+	// ErrOracleFinished represents SubmitError with code -602.
+	ErrOracleFinished = NewSubmitError(-602, "Oracle request has already been finished")
+	// ErrOracleNotFound represents SubmitError with code -603.
+	ErrOracleNotFound = NewSubmitError(-603, "Oracle request is not found")
+	// ErrOracleNotDesignatedNode represents SubmitError with code -604.
+	ErrOracleNotDesignatedNode = NewSubmitError(-604, "Not a designated oracle node")
+
+	// ErrOldState represents SubmitError with code -701.
+	ErrOldState = NewSubmitError(-701, "Old state requests are not supported")
+	// ErrProof represents SubmitError with code -702.
+	ErrProof = NewSubmitError(-702, "Invalid proof")
+
+	// ErrNotarySvcNotRun represents SubmitError with code -801.
+	ErrNotarySvcNotRun = NewSubmitError(-801, "Notary service is not running")
+	// ErrNotaryFallback represents SubmitError with code -802.
+	ErrNotaryFallback = NewSubmitError(-802, "Invalid fallback transaction sender")
+	// ErrNotaryUnableToVerify represents SubmitError with code -803.
+	ErrNotaryUnableToVerify = NewSubmitError(-803, "Fallback transaction is valid after deposit is unlocked")
+	// ErrNotaryNode represents SubmitError with code -804.
+	ErrNotaryNode = NewSubmitError(-804, "Not a designated notary node")
+	// ErrNotarySignature represents SubmitError with code -805.
+	ErrNotarySignature = NewSubmitError(-805, "Invalid notary request signature")
 )
 
```

Comment:
Extending our `Error` with `Code() int64` can be useful as well.

Reply:
Do you mean usage like this: `ErrNotarySignature.Code()`? It works